src/iotjs_binding.h

```diff
@@ -216,4 +219,37 @@ jerry_value_t iotjs_jhelper_eval(const char* name, size_t name_len,
 
 jerry_value_t vm_exec_stop_callback(void* user_p);
 
+/**
+ * Dynamic module defintions (.iotjs)
+ */
+
+#define IOTJS_MAKE_VERSION(major, minor, patch)               \
+  (((0x10 & 0xFF) << (3 * 8)) | ((major & 0xFF) << (2 * 8)) | \
+   ((minor & 0xFF) << (1 * 8)) | (patch & 0xFF))
+
+#define IOTJS_GET_VERSION_MAJOR(version) ((version >> (2 * 8)) & 0xFF)
+#define IOTJS_GET_VERSION_MINOR(version) ((version >> (1 * 8)) & 0xFF)
+#define IOTJS_GET_VERSION_PATCH(version) (version & 0xFF)
+
+typedef jerry_value_t (*ModuleInitializer)(void);
+
+typedef struct {
+  uint32_t iotjs_version;
+  uint32_t module_version;
+  uint32_t flags; /* reserverd for future use */
+  char name[255];
+  ModuleInitializer initializer;
+} iotjs_module;
+
+typedef iotjs_module* (*iotjs_module_info_getter)(void);
```

Comment:
The negative side is that the structure must be static. Btw if the structure is static, the name could be `char*` above.

Reply:
If you use (and you should really use) the macro `IOTJS_MODULE` below, the static struct will be created correctly. I'm not against having the name as a `char*`